crates/taplo-common/src/schema/mod.rs

```diff
@@ -134,52 +136,53 @@ impl<E: Environment> Schemas<E> {
         // a validation path that requires it, so we might have to loop many times
         // to fully validate according to a schema that has many nested references.
         loop {
-            match validator.validate(value) {
-                Ok(()) => return Ok(Vec::new()),
-                Err(errors) => {
-                    let errors: Vec<_> = errors
-                        .map(|err| ValidationError {
-                            instance: Cow::Owned(err.instance.into_owned()),
-                            kind: err.kind,
-                            instance_path: err.instance_path,
-                            schema_path: err.schema_path,
-                        })
-                        .collect();
-
-                    // We check whether there were any external schema errors,
-                    // and retrieve the schemas accordingly.
-                    let mut external_schema_requests: FuturesUnordered<_> = errors
-                        .iter()
-                        .filter_map(|err| {
-                            if let ValidationErrorKind::Resolver { url, .. } = &err.kind {
-                                Some(async {
-                                    let value = self.load_schema(url).await?;
-                                    drop(self.cache.store(url.clone(), value));
-                                    Result::<(), anyhow::Error>::Ok(())
-                                })
-                            } else {
-                                None
-                            }
+            let errors = validator
+                .iter_errors(value)
+                .map(|err| ValidationError {
```

Comment:
Would `ValidationError::to_owned` work here? It does the same (except an extra clone that I've missed to remove)